src/firmware_new/boards/frankie_v1/io/allegro_a3931_motor_driver.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+/**
+ * This file is an abstraction around the Allegro A3931 Motor Driver
+ */
+
+#include "firmware_new/boards/frankie_v1/io/gpio_pin.h"
+#include "firmware_new/boards/frankie_v1/io/pwm_pin.h"
+#include "stm32h7xx_hal.h"
+
+typedef enum
+{
+    CLOCKWISE,
+    COUNTERCLOCKWISE
+} AllegroA3931MotorDriverDriveDirection;
+
+typedef struct AllegroA3931MotorDriver AllegroA3931MotorDriver_t;
+
+/**
+ * Create a Motor Driver
+ * @param pwm_pin The PWM pin to the driver. Note that this must be configured to
+ *                operate in the correct frequency range for the motor driver. Please
+ *                talk to Elec. to determine this.
+ * @param reset_pin A GPIO pin connected to the "reset" pin on the motor driver.
+ * @param coast_pin A GPIO pin connected to the "coast" pin on the motor driver.
+ * @param mode_pin A GPIO pin connected to the "mode" pin on the motor driver.
+ * @param direction_pin A GPIO pin connected to the "direction" pin on the motor driver.
+ * @param brake_pin A GPIO pin connected to the "brake" pin on the motor driver.
+ * @param esf_pin A GPIO pin connected to the "ESF" pin on the motor driver.
+ *
+ * @return A motor driver.
+ */
+AllegroA3931MotorDriver_t* io_allegro_a3931_motor_driver_create(
+    PwmPin_t* pwm_pin, GpioPin_t* reset_pin, GpioPin_t* coast_pin, GpioPin_t* mode_pin,
+    GpioPin_t* direction_pin, GpioPin_t* brake_pin, GpioPin_t* esf_pin);
+
+/**
+ * Destroy the given motor driver
+ * @param motor_driver The motor driver to destroy
+ */
+void io_allegro_a3931_motor_driver_destroy(AllegroA3931MotorDriver_t* motor_driver);
+
+/**
+ * Set the rotation direction for the given motor driver
+ *
+ * Note that the rotation is from the perspective of rear of the motor, looking out
```

Comment:
I think `from the perspective of inside the robot, looking along the motor shaft` might be more clear? "rear of the motor" sounds more confusing IMO.

Definitely want more thoughts on this @akhilveeraghanta @EvanMorcom

Reply:
I thought of that, but what if this is motor driver is connected to the dribbler motor?

Reply:
oh good point. My only other idea is "the base of the axle" but I don't think that's necessarily better. Fine with "rear" as-is since I can figure it out reasonably quickly.

Reply:
'Down the shaft towards the wheel'

Reply:
What if it's the dribbler motor?

Reply:
'Down the shaft towards the gearing'

Reply:
How about `Down the shaft starting from the motor body`?

Reply:
don't see this change yet

Reply:
Was just waiting until we resolved it here. Changed and pushed.